src/net/wooga/jenkins/pipeline/BuildUtils.groovy

```diff
@@ -0,0 +1,56 @@
+package net.wooga.jenkins.pipeline
+
+import org.jenkinsci.plugins.pipeline.modeldefinition.Utils
+
+class BuildVersion {
+    final String version
+    final Boolean optional
+
+    BuildVersion(version, optional) {
+        this.version = version
+        this.optional = optional
+    }
+    
+    @Override
+    String toString() {
+        return "version : ${version}, optional : ${optional}"
+    }
+}
+
+// For each entry in a versions map, if they are not in the standard format
+// we process them into the newer format that supports optional arguments per version
+// (such as if it's optional, etc)
+// Example: '3001.x' -> [ version : '3001.x', optional : false ]
+def parseVersions(versions) {
+
+    final versionKey = "version"
+    final optionalKey = "optional"
+
+    versions = versions.collect { $it -> 
+    
+        def v = $it
+        
+        // If the entry is a map, validate that it has the fields we care for
+        if (v instanceof Map) {
+            if (!v.containsKey(versionKey)){
+                throw new Exception("Entry ${v} does not contain ${versionKey}")
+            }
+            if (!v.containsKey(optionalKey)){
+                throw new Exception("Entry ${v} does not contain ${optionalKey}")
+            }
+            return new BuildVersion(v[versionKey], v[optionalKey])
+            //return v
+        }
+        
+        // If only a string was passed, convert to the version data object
+        else if (v instanceof String) {            
+            return new BuildVersion(v, false)
+            //return [ "${versionKey}" : v, "${optionalKey}" : false ]
+        }
```

Comment:
I would instead call `new BuildVersion(v.toString(), false)` as the fallback case as we never defined what the version string should look like. Which is ok I think.

Reply:
Got it.